### PR TITLE
docs(toh-6): add hero search to Dart; minor edits to TS

### LINKED ISSUE #1924: [Dart] ToH-6 updates following addition of Observables section on TS side
Dart follow-up to #1528.

### PR DESCRIPTION
**Pull request by chalin**
TS minor edits:
- @wardbell @Foxandxss : as usual, [TS side edits are best viewed by **ignoring whitespace**](https://github.com/angular/angular.io/commit/94d432d4740fe58205aacd009549b3cb3af873e1?w=1).

Dart update:
- @kwalrath @thso: fixes #1924.

Note: in anticipation of #1751, `sample.css` has been eliminated from the Dart example.

### CODE REVIEW

File: public/docs/ts/latest/tutorial/toh-pt6.jade [hunk 1]
```diff
@@ -135,21 +135,21 @@ block get-heroes-details
 
     For *now* we get back on familiar ground by immediately by 
     converting that `Observable` to a `Promise` using the `toPromise` operator.
-  +makeExample('toh-6/ts/app/hero.service.ts', 'to-promise')(format=".")
+  +makeExcerpt('app/hero.service.ts', 'to-promise')
```
Comment (kwalrath) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: I'm not crazy about adding the big blue title to all the one-line examples. It gives the source of the code more weight than the actual code.

File: public/docs/ts/latest/tutorial/toh-pt6.jade [hunk 2]
```diff
@@ -384,59 +385,78 @@ block observables-section
...
-    As the user types a name into a search box, we'll make repeated http requests for heroes filtered by that name.
+  ### Search-by-name
+  We're going to add a *hero search* feature to the Tour of Heroes.
+  As the user types a name into a search box, we'll make repeated HTTP requests for heroes filtered by that name.
 
-    We start by creating `HeroSearchService` that sends search queries to our server's web api.
+  We start by creating `HeroSearchService` that sends search queries to our server's web api.
 
-  +makeExample('toh-6/ts/app/hero-search.service.ts', null, 'app/hero-search.service.ts')(format=".")
++makeExample('app/hero-search.service.ts')
```
Comment (kwalrath) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: Interestingly, the version with the line numbers is vertically more compact (in a good way) than the original version.

File: public/docs/ts/latest/tutorial/toh-pt6.jade (hunk 1 above)
Comment (chalin) [inline public/docs/ts/latest/tutorial/toh-pt6.jade]: Oups, didn't mean for that to happen. I forgot that the convention for including/excluding a title with `makeExcerpt` is reversed as compared to `makeExample`. Fixed all of excerpts that originally did not have a title.

**Comment by chalin** [comment]
@kwalrath : updates pushed.

File: public/docs/dart/latest/tutorial/toh-pt6.jade [hunk 3]
```diff
@@ -89,56 +96,108 @@ block heroes-comp-add
...
+    the stream by calling `add` over the controller.
+
+    [Stream]: https://api.dartlang.org/stable/dart-async/Stream-class.html
+    [StreamController]: https://api.dartlang.org/stable/dart-async/StreamController-class.html
+
+block observable-transformers
+  :marked
+    Fortunately, there are stream transformers that will help us reduce the request flow.
+    We'll make fewer calls to the `HeroSearchService` and still get timely results. Here's how:
+    starting from the underlying stream of search terms we
```
Comment (kwalrath) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: This line looks weird and ungrammatical. Can we just delete it?

File: public/docs/dart/latest/tutorial/toh-pt6.jade [hunk 4]
```diff
@@ -89,56 +96,108 @@ block heroes-comp-add
...
+
+    * Debounce for 300 milliseconds, that is, wait until the flow of search terms pauses for the
```
Comment (kwalrath) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: The TS version of this mentions the code/API that does each thing. I rather like that. It tells you where to look in the code up above.

File: public/docs/dart/latest/tutorial/toh-pt6.jade [hunk 5]
```diff
@@ -89,56 +96,108 @@ block heroes-comp-add
...
+    specified time. This way, we'll never make requests more frequently than every 300ms.
+
+    * Skip repeated strings, ensuring that we only send a request if a search term has changed.
+    There's no point in repeating a request for the same search term.
+
+    * Apply a map-like transformer which: (1) calls our search service for each search term that
```
Comment (kwalrath) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: which: -> that

(no semi-colon, clarify that this is a restrictive clause)

File: public/docs/dart/latest/tutorial/toh-pt6.jade [hunk 6]
```diff
@@ -89,56 +96,108 @@ block heroes-comp-add
...
+    makes it through the debounce and distinct gauntlet; (2) returns only the most recent search
```
Comment (kwalrath) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: ; -> and

**Comment by kwalrath** [comment]
Some minor issues, but otherwise looks great!

File: public/docs/dart/latest/tutorial/toh-pt6.jade (hunk 4 above)
Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: IMHO, the correspondence is easy to make. The problem with the Dart is that some of the commands are quite verbose, and it isn't clear how to refer to them. E.g. TS
- `debounceTime(300)`

vs. Dart
- `transform(new Debounce(new Duration(milliseconds: 300)))`

Unless you find an abbreviation like
- `transform(new Debounce(... 300)))`

acceptable. Let me know what you prefer.

File: public/docs/dart/latest/tutorial/toh-pt6.jade (hunk 3 above)
Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: Deleted.

File: public/docs/dart/latest/tutorial/toh-pt6.jade (hunk 5 above)
Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: Fixed.

Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: Oh, did you mean no colon?

File: public/docs/dart/latest/tutorial/toh-pt6.jade (hunk 6 above)
Comment (chalin) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: Done.

File: public/docs/dart/latest/tutorial/toh-pt6.jade (hunk 4 above)
Comment (kwalrath) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: Abbreviations are great.

File: public/docs/dart/latest/tutorial/toh-pt6.jade (hunk 5 above)
Comment (kwalrath) [inline public/docs/dart/latest/tutorial/toh-pt6.jade]: yep, that's what I meant. :)

**Comment by kwalrath** [comment]
LGTM

**Pull request merged**